### ISSUE TITLE
deal.II Release 9.0 5/11/2018

### ISSUE THREAD

**Issue by tamiko** [dealii/dealii#6156]
[deal.II version 9.0.0](https://github.com/dealii/dealii/releases/tag/v9.0.0)
[Documentation version v9.0.0](https://tamiko.kyomu.43-1.org/tmp/doc/)
====

Open tasks:
---
- Update dealii in [Candi](https://github.com/dealii/candi/blob/master/candi.cfg)
- Update dealii in [Ubuntu/Debian](https://salsa.debian.org/science-team/deal.ii) (Matthias)
- Update dealii virtualbox images (Timo)
- Send a note to the Facebook and Google+ accounts (Wolfgang/Timo)

Merge history:
---
 - final: e9f6e3ab9d495d9c8a14b11e2d4a374b6acc6fa1
    - #6580 (take over #6567)
    - #6579 (take over #6577)
    - #6575 (take over #6573, #6574)
    - #6571 (take over #6569)
    - #6570 (take over #6564)
    - #6563 (take over #6542)
    - #6561 (take over #6555)
    - #6560 (take over #6553)
 - rc5: 6ac831ab500cc6eca5cf66b9eaad47419d2a226a
    - #6551 (take over #6540)
 - rc4: 27f62c0b80
    - #6549 (take over #6544, #6543, #6541)
    - #6548 (take over #6537, #6538
    - #6547 (take over #6535, #6536)
    - #6534 (take over #6533)
    - #6521 (take over #6297)
    - #6516 (take over #6291, #6511, #6513)
    - #6515 (take over #6508, #6519)
    - #6514 (take over #6393)
 - rc3: 3975bada23997cdc376a247bd55295a4eedbfca8
    - #6509 (take over #6475)
    - #6507 (take over #6499)
    - #6506 (take over #6498)
    - #6505 (take over #6496)
    - #6504 (take over #6495)
    - #6503 (take over #6491)
    - #6502 (take over #6489)
    - #6501 (take over #6483)
    - #6500 (take over #6480)
    - #6492
    - #6485
 - rc2: bfe525dc113b40eeea79a197aa5eebaac9566f1f
    - #6482
    - #6479 (take over #6474)
    - #6478 (take over #6476)
    - #6477 
 - rc1: 98208d44b01e6b60cc31ee08514633300c11fcfa

Testsuite results for v9.0.0-rc5-6-ge549f8f0bf e549f8f0bf
---
 |  Host            |  Configuration                                                                                         |  Commit                                                            |  Build errors  |  Build warnings                                                                        |  Failing tests                                                                        |  Passing tests  |  
 |  -               |  -                                                                                                     |  -                                                                 |  :-:           |  :-:                                                                                   |  :-:                                                                                  |  :-:            | 
 |  tester          |  [GNU-7.3.0-dealii-9.0-unity_build](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1660)        |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  0              | 
 |  tester          |  [GNU-7.3.0-dealii-9.0-all_components](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1661)     |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  0              | 
 |  tester          |  [GNU-7.3.0-dealii-9.0-minimal_bundled](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1662)    |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  7090           | 
 |  tester          |  [GNU-4.9.4-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1666)      |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  10093          | 
 |  tester          |  [GNU-6.4.0-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1668)      |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  10157          | 
 |  tester          |  [GNU-7.3.0-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1673)      |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  10157          | 
 |  tester          |  [GNU-8.1.0-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1675)      |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  10157          | 
 |  tester          |  [Clang-4.0.1-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1679)    |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  10139          | 
 |  tester          |  [Clang-5.0.1-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1681)    |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  10139          | 
 |  tester          |  [Clang-6.0.0-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1683)    |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  0                                                                                    |  10139          | 
 |  tester          |  [GNU-7.3.0-dealii-9.0-petsc_complex](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1684)      |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  [<b>1</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1684)    |  9850           | 
 |  tester          |  [GNU-7.3.0-dealii-9.0-64bit_indices](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1686)      |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  [<b>4</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1686)    |  10119          | 
 |  tester          |  [GNU-7.3.0-dealii-9.0-optimized](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1689)          |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  [<b>20</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1689)   |  10137          | 
 |  tester          |  [Clang-6.0.0-dealii-9.0-optimized](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1690)        |  [e549f8f0bf](https://github.com/dealii/dealii/commit/e549f8f0bf)  |  0             |  0                                                                                     |  [<b>11</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1690)   |  10128          | 
 |  simserv04       |  [GNU-5.3.0-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1674)      |  [efe2dd0](https://github.com/dealii/dealii/commit/efe2dd0)        |  0             |  0                                                                                     |  0                                                                                    |  9051           | 
 |  simserv04       |  [Clang-3.7.0-dealii-9.0-autodetection](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1677)    |  [efe2dd0](https://github.com/dealii/dealii/commit/efe2dd0)        |  0             |  0                                                                                     |  [<b>2</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1677)    |  9049           | 
 |  simserv02       |  [Intel-18.0.1.20171018](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1680)                   |  [e549f8f](https://github.com/dealii/dealii/commit/e549f8f)        |  0             |  0                                                                                     |  [<b>4</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1680)    |  8405           | 
 |  simserv02       |  [Intel-19.0.0.20180317](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1682)                   |  [e549f8f](https://github.com/dealii/dealii/commit/e549f8f)        |  0             |  [<b>19</b>](https://cdash.kyomu.43-1.org/viewBuildError.php?type=1&amp;buildid=1682)  |  [<b>115</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1682)  |  8316           | 
 |  davyddenubuntu  |  [GNU-7.3.0-dealii-9.0-^openmpi^openblas](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1687)  |  [efe2dd00](https://github.com/dealii/dealii/commit/efe2dd00)      |  0             |  0                                                                                     |  0                                                                                    |  10147          | 
 |  CUDA8           |  [GNU-5.4.0-dealii-9.0](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1671)                    |  [efe2dd0](https://github.com/dealii/dealii/commit/efe2dd0)        |  0             |  [<b>80</b>](https://cdash.kyomu.43-1.org/viewBuildError.php?type=1&amp;buildid=1671)  |  [<b>2</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1671)    |  7068           | 
 |  CUDA9           |  [GNU-5.4.0-dealii-9.0](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1672)                    |  [efe2dd0](https://github.com/dealii/dealii/commit/efe2dd0)        |  0             |  0                                                                                     |  [<b>2</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1672)    |  7068           | 
 |  node27          |  [GNU-6.1.0-dealii-9.0](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1676)                    |  [efe2dd0](https://github.com/dealii/dealii/commit/efe2dd0)        |  0             |  0                                                                                     |  [<b>3</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1676)    |  9183           | 
 |  node27          |  [GNU-6.1.0-dealii-9.0-nothread](https://cdash.kyomu.43-1.org/buildSummary.php?buildid=1678)           |  [efe2dd0](https://github.com/dealii/dealii/commit/efe2dd0)        |  0             |  0                                                                                     |  [<b>3</b>](https://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&amp;buildid=1678)    |  9163           | 

Checklist for [release-tasks](https://github.com/dealii/release-papers/blob/master/release-tasks)
===

Release preparation steps:
---

- [x] <b>-3</b>  Send out notice to the mailinglist if anyone has outstanding issues or patches.
- [x] <b>-2</b>  Check all issues and pull-request on github. Create a list of things to fix before the release.
- [x] <b>-1</b>  Update the list of publications. (Send standard e-mail)
- [x] <b>-1a</b> Start release paper process with email to all current (principal) developers
- [x] <b>-1b</b> Update the authors list on the homepage authors.html
- [x] <b>-1c</b> Update the list of contributors in the release paper
- [x] <b>0</b>   Do some basic testing as always
- [x] <b>0a</b>  Rerun cppcheck across the source files
- [x] <b>0b</b>  Check for doxygen formatting problems
- [x] <b>0c</b>  Fix formatting bugs reported by various command line queries
- [x] <b>0d</b>  Check that doxygen produces no errors when generating offline docs.
- [x] <b>0e</b>  Run the copyright script
- [x] <b>0f</b>  Update Trilinos and PETSc to the latest release on the tester and update documentation of supported version.
- [x] <b>0g</b>  Run codespell

Release steps:
---

- [x] <b>*</b>   Nominate a "release officer" that is in charge of (almost all) of the following steps
- [x] <b>1</b>   Create new changes file
- [x] <b>1a</b>  Adjust doc/users/doxygen.html to link to the to-be-created TAG file.
- [x] <b>4</b>   Create a release branch
- [x] <b>5</b>   Update the VERSION from 9.0.0-pre to 9.1.0-pre on master
- [x] <b>6</b>   Update the official tester to test on the release branch dealii-9.0
- [x] <b>6a</b>  On mainline, update deal.II dependencies of examples
- [x] <b>7</b>   Create an AUTHORS file
- [x] <b>7a</b>  Send prerelease announcement to dealii@googlegroups.com
- [x] <b>8</b>   Remove unfinished tutorial programs.
- [x] <b>9</b>   Write up a short release note and save it as announce-9.0
- [x] <b>9a</b>  Write the release paper
- [x] <b>10</b>  Test the branch thoroughly
- [x] <b>10a</b> Do the same for the code gallery examples
- [x] <b>11</b>  Create documentation and verify
- [x] <b>12</b>  Push the button
    - Change VERSION to 9.0.0 for a release and to 9.0.0-rc1 for a pre-release
    - Create a signed tag of exactly this commit
    - Push changes to github
- [x] <b>13</b>  Create the tar files
    - Download the source tarball from github, verify that its contents is what we expect
    - Cryptographically sign it (detached, armored signature)
    - Create full offline_doc tarball and sign it
- [x] <b>14a</b> Create a pre-release on github and upload files (and detached signatures)
- [x] <b>15</b>  Generate the documentation on the webserver.
- [x] <b>16</b>  Adjust header.include on homepage
- [x] <b>17a</b> Ask Luca for Mac packages
- [x] <b>17b</b> Update dealii in [Spack](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/dealii/package.py)
- [ ] <b>17c</b> Update dealii in [Candi](https://github.com/dealii/candi/blob/master/candi.cfg)
- [x] <b>17d</b> Update dealii in [Gentoo Science overlay](https://gitweb.gentoo.org/proj/sci.git/tree/sci-libs/dealii)
- [ ] <b>17e</b> Update dealii in [Ubuntu/Debian](https://salsa.debian.org/science-team/deal.ii)
- [x] <b>17f</b> Update dealii in [Archlinux AUR](https://aur.archlinux.org/packages/deal-ii/)
- [ ] <b>17g</b> Update dealii virtualbox images
- [x] <b>18</b>  Finish writing the announcement
- [x] <b>18b</b> Adjust news.html in the deal.II homepage repo
- [x] <b>19</b>  Announce the release via e-mail
- [ ] <b>19a</b> Send a note to the Facebook and Google+ accounts
- [x] <b>19b</b> Update the deal.II wikipedia page
- [x] <b>19d</b> Send thank-you email and list the people already referenced in step -1b above
- [x] <b>20</b>  Celebrate!

**Comment by drwells** [comment]
Do we have a deadline for when we will create a release branch and only accept bugfixes?

**Comment by davydden** [comment]
End of April?

**Comment by kronbichler** [comment]
I guess @drwells also commented on being more conservative on master before creating the branch. In my opinion it would make sense to have that deadline a bit before April 27. I would have said that all non-bugfix PRs should be posted by April 20 to have some time to review and discuss. But even that seems ambitious if we target a "release". On the other hand, we have a few substantial issues left on our [list](https://github.com/dealii/dealii/milestone/5). We should decide this week which of those we want to finish and which ones to postpone because it seems unrealistic to address all 30 open issues for the 9.0 release.

**Comment by bangerth** [comment]
Can we switch to a mode where we now only accept bug fixes, doc changes, and, possibly, new tutorial programs? We really need to get to a point where we can cut the release branch.

What functionality is currently still pending? What is lacking regarding the manifold changes that still needs to go in?

**Comment by bangerth** [comment]
Also, if everyone could go through the milestone list at https://github.com/dealii/dealii/milestone/5 and think about whether their favorite issue really needs to be in 9.0 or could slip to 9.1, that would be great!

**Comment by kronbichler** [comment]
> Can we switch to a mode where we now only accept bug fixes, doc changes, and, possibly, new tutorial programs? We really need to get to a point where we can cut the release branch.

I have #5667 which is 75% through. I will post the second to last big patch (everything apart from `FEFaceEvaluation`) tonight. I apologize for being late, but it was a lot to review and we did our best to break things down which delayed things since we started in February.

**Comment by Rombur** [comment]
I would like to merge one more PR for cuda. It will be ready tonight, the code is ready but the computer were the code is, was down during the week-end so I couldn't create the PR.

**Comment by tamiko** [comment]
What about we fork off the release branch this Friday and create a first release candidate? This should give everyone enough time for their last-minute inclusions :-)

After that I suggest to tag the second release candidate Friday in a week. Depending on how things calmed down in that week we can make the final releas shortly after :-)

**Comment by tamiko** [comment]
I have compiled a handy check list for our release tasks (see updated first post).
As always, we need exactly 20 steps for a release ;-)

**Comment by tjhei** [comment]
> As always, we need exactly 20 steps for a release ;-)

:-)

**Comment by masterleinad** [comment]
> As always, we need exactly 20 steps for a release ;-)

Actually, 42 is a pretty nice number.

**Comment by bangerth** [comment]
@tamiko -- thanks for converting the release tasklist file into this github issue. This is more easy to work with. If you wanted to convert the file in a way so that next time we only need to copy-paste it into a github issue, then that would be ok as well!

**Comment by tamiko** [comment]
Depending on how much work we get done over the course of this day I plan to cut a release branch at around midnight UTC (that is 19:00 Central time).

**Comment by luca-heltai** [comment]
There is still step-60 (missing basically only the results section) and a gmsh function I intended to add.

**Comment by luca-heltai** [comment]
If you push for a release *today*, that's not going to happen on my side, unless @GivAlz manages to get things into a final shape in step-60 by midnight your time.

**Comment by tamiko** [comment]
Well, then it is in a week, I guess.

**Comment by tamiko** [comment]
@luca-heltai Actually, I wanted to tag a pre-release, not a release. But let's wait with that for another week.

**Comment by luca-heltai** [comment]
If I'm the only one holding back, by all means let's call it a day, and go for a pre-release today. 

I thought step-60 would have been a good thing to have, since it basically uses only new functionality, not treated in any other tutorial (MappingFEField, NonMatching, ParameterAcceptor, and others...)

**Comment by tamiko** [comment]
@luca-heltai We can also tag a release today (rc1) and merge the new tutorial step into rc2 next week Friday. That way we would have a feature-frozen release candidate for testing and you would have a week to finish the example step. I can also help Tuesday onwards if you wish :-)

**Comment by luca-heltai** [comment]
Let's wait before we remove gmsh though...

**Comment by kronbichler** [comment]
@kkormann and I can get that tutorial done until next week, but we need half a day (which is first next Wednesday for me) because we have not written the documentation yet, just the plain program. Let us see what others think about this.

**Comment by davydden** [comment]
> @kkormann and I can get that tutorial done until next week

IMO one week does not break the deal (pun intended) 😄  so let's more-or-less freeze the features but not branch off yet as we have a couple of opened issues.

**Comment by GivAlz** [comment]
One week would be great: it would allow to finish the result section of step-60 and improve the tutorial overall! I noticed the NonMatching functions need a better documentation, which I would also write as it is functional for the tutorial! Thanks :)

**Comment by jppelteret** [comment]
This would also give me time to attend to #6172, so I vote with the crowd on this.

**Comment by tamiko** [comment]
Guess next week it is then :-)

**Comment by bangerth** [comment]
It takes a substantial amount of time to go through the release task list. There is value to making the branch earlier rather than later, so that people can go through the list. We can always pull commits over to the branch.

Could I advocate for the following:
* We work on merging all of the pending bug fix and doc fix PRs.
* When they are in, we cut the branch (whenever that is).
* We go through the release task list and do all of the work on the release branch as necessary.
* If one or both of the tutorial programs happens to be ready before the release date, we can always pull that over without too much risk.

So the work for the moment is that we need to go through the open pull requests and issues that we know we *need* to get through.

**Comment by tamiko** [comment]
@bangerth 
> - We work on merging all of the pending bug fix and doc fix PRs.
> - When they are in, we cut the branch (whenever that is).

I strongly agree. Further, I suggest, we do not merge anything else to master until we forked off.

**Comment by kronbichler** [comment]
I also agree.

I have to fix a few failing tests due to #6300 (because SSE generates different output than AVX that I tested with) that I will commit later today: http://cdash.kyomu.43-1.org/viewTest.php?onlyfailed&buildid=1176

**Comment by tamiko** [comment]
All, as far as I can see we are basically down to
 - a number of minor and major documentation updates that are still partially pending
   (https://github.com/dealii/dealii/issues/6274, https://github.com/dealii/dealii/issues/6184, https://github.com/dealii/dealii/issues/6183)
 - one obscure bug (https://github.com/dealii/dealii/issues/6229)
 - Two pending pull requests (https://github.com/dealii/dealii/pull/6434, https://github.com/dealii/dealii/pull/6428)
 - Two new example steps (https://github.com/dealii/dealii/pull/6291, https://github.com/dealii/dealii/pull/6297)

If everyone is OK with it, I would like to:
 - suggest to embargo the main branch for the next 24h (only pull requests for above stuff gets merged).
 - tag a prerelease tomorrow at above time immediately after the two pending pull requests are merged.
 - suggest we merge further documentation fixes on the release branch as they come in (over the next week).
 - suggest we wait for the actual release for step-59 and step-60 to be polished and merged to the release.

I would really like to freeze the library right now with most of the testsuite being exceptionally happy...
I don't see any problems for documentation and example steps being merged quite late in the release cycle.

**Comment by tamiko** [comment]
I have nominated e4483a2 for forking off. I will run bob with one configuration quickly to make sure that everything is in order.

**Comment by drwells** [comment]
I posted a warning in the gitter chat that I should probably repeat here.

A possible problem with 9b5030eba6bf6b536037dab8e7d7a7f124fa3866 is that it changes (slightly) the 96 cell hyper shell in 3D. We have about 50 tests that rely on this grid, so I may have broken more tests due to this.

I'm looking into it now (I am rerunning the full test suite on my workstation).

**Comment by tamiko** [comment]
@drwells I will wait with the branching until after the testsuite results.

**Comment by tamiko** [comment]
@dealii/dealii We have a release branch now \o/ I will tag a RC1 sometime tomorrow after the latest fallout (https://github.com/dealii/dealii/issues/6458) is fixed.

**Comment by tjhei** [comment]
> the 96 cell hyper shell in 3D

this will have additional fall-out for ASPECT and its testsuite @bangerth @gassmoeller

**Comment by gassmoeller** [comment]
Depends, how confident are you that the new version is better? We do not check for correctness of test results with the recent deal.II dev (our reference tester is running 8.5) so there is no inherent reason not to change something for deal.II 9.0, just because it changes things for aspect. Of course it would be bad if we later on discover the new version is worse than the existing one, but at least in the result that you show above the lower iteration count suggests the new geometry is actually better (and I get the same diff on my system).

**Comment by tamiko** [comment]
@drwells I wouldn't be too worried about the hypershell update. We also have a week to assess how much fallout this created.

**Comment by tamiko** [comment]
@masterleinad @Rombur @davydden Please reconfigure all testers that report to `Continous` to test tag v9.0.0-r1, commit  98208d4 so that we can update above table :-)

**Comment by davydden** [comment]
Will do on Monday

**Comment by drwells** [comment]
@gassmoeller I did not examine the quality of the shapes: they are different by more than roundoff errors but yes, the new points are computed by SphericalManifold.

**Comment by drwells** [comment]
I think the new grid is better: I posted a longer discussion in the aspect tracker.

**Comment by tamiko** [comment]
@davydden Wise choice - v9.0.0-rc2 it should be by then ;-)

**Comment by tamiko** [comment]
@masterleinad @Rombur @davydden Please run all testing sites reporting to `Continuous` once on branch `dealii-9.0`, specifically commit 3975bad

**Comment by Rombur** [comment]
The current status is that we have problems with the testing machine :)

**Comment by kronbichler** [comment]
> Please run all testing sites reporting to Continuous once on branch dealii-9.0, specifically commit 3975bad

The last three characters make this an extremely funny hash name for a release tag... :laughing:

**Comment by tamiko** [comment]
@kronbichler Jup!

**Comment by davydden** [comment]
> Please run all testing sites reporting to Continuous once on branch dealii-9.0

my `openmpi+openblas` tester only complained about `gmsh/create_tria_02`, which is failing everywhere... I guess @bangerth was right that we should have postponed the new functionality ;-(

**Comment by tamiko** [comment]
> my `openmpi+openblas` tester only complained about
> `gmsh/create_tria_02`, which is failing everywhere... I guess
> @bangerth was right that we should have postponed the new
> functionality ;-(

The test and the functionality is fixed on master. I have high hopes for
rc4 to be down to 0 failing tests and no known regressions. :-)

**Comment by drwells** [comment]
Do we have an ETA for rc4?

**Comment by kronbichler** [comment]
@tamiko #6291 and #6511 are already merged to master. It's now only a tiny update #6513 that is missing for step-59.

**Comment by drwells** [comment]
I don't know how close step-60 is so I do not have a strong opinion on deadlines (other than we should have both new steps included).

**Comment by davydden** [comment]
> I don't know how close step-60 is

As far as I can tell, all the comment are addressed, @jppelteret wanted to give it another look as he previously requested changes. I think tagging rc4 on Wednesday and releasing on Friday is doable.

**Comment by jppelteret** [comment]
Yip, I've reviewed #6297 and I know that I've still got some documentation outstanding. The (now more concrete) plan for me is to finish #6172 tomorrow. Hopefully @drwells and I can come to some sensible resolution on #6473 as well.

**Comment by bangerth** [comment]
If #6172 and #6437 don't happen today, I vote for postponing to 9.1. Both fall in the "nice to have, but not critical" category.

Re @davydden 's comment about the `mkstemp` issue -- we'll have to have some sort of debrief about our release process once we're finally done.

**Comment by tamiko** [comment]
@bangerth I thought we already postponed #6437 to 9.1, or should I have worked on that for 9.0?!

**Comment by davydden** [comment]
> I thought we already postponed #6437 to 9.1, or should I have worked on that for 9.0?!

I think he meant #6473  (last two numbers swapped)

**Comment by tamiko** [comment]
@davydden That makes much more sense. :-)

**Comment by bangerth** [comment]
Bizarre that both numbers made sense. And yes, I swapped the numbers.

**Comment by tamiko** [comment]
@bangerth Above is my suggestion for a final set of merges for rc4. I think we shouldn't do much more.

**Comment by davydden** [comment]
we also have those two https://github.com/dealii/dealii/issues/6274 and https://github.com/dealii/dealii/issues/6183 marked with 9.0 milestone.

**Comment by bangerth** [comment]
Let's skip #6517. It's a nuisance that it doesn't work, but I don't think it's critical and we need to come to a conclusion that doesn't require us running the test suite again and again. It can wait for 9.1.

So the 4 pull requests you mention sound good to me.

**Comment by bangerth** [comment]
Let's postpone #6183 (glossary entry for distorted cells) to 9.1 -- same story: nice to have, not critical. (It's going to be outdated -- not ideal, but we can live with that). 

I'd say let's also postpone whatever the remainder of #6274 (documentation of `GridGenerator`) is. It looks like almost everything was updated. Whatever else is left are probably marginal cases where either the documentation is not quite up to date or where we'd have to change code in non-trivial ways (with the consequent need to run tests). But I'd like to let @drwells make that call.

**Comment by drwells** [comment]
I would like to attach `TorusManifold` objects by default. This is not our most popular GridGenerator function but I believe it is the last one whose behavior needs to be updated.

**Comment by tamiko** [comment]
@dealii/dealii All, I will restart `tester` on the release branch now. If all is going well (i.e. the first 4 configuration pass), I pan to tag `rc4` at around 1pm central time and restart again.
Have I missed any urgent last minute changes that should go into `rc4`?

**Comment by tamiko** [comment]
@dealii/dealii *mhm* I am actually tempted to wait a bit for #6537, #6538, #6536, #6535 and add them to the rc4 as well. (And after that bite at everyone who wants to have further PRs that are not critical bugfixes ^^). Opinions?

**Comment by drwells** [comment]
I will, of course, lobby for #6537 and #6538. The other two adjust tests and should also be okay.

**Comment by tamiko** [comment]
@luca-heltai What version of `gmsh` do you use and what features are configured? I cannot get the `gmsh/create_tria_02` test to pass on any of my machines.

**Comment by luca-heltai** [comment]
634-heltai@turbolo: step-60[hot-fix-step-60] $ gmsh -info
Version          : 3.0.6
License          : GNU General Public License
Build OS         : MacOSX
Build date       : 20180505
Build host       : turbolo
Build options    : 64Bit Ann Bamg Bfgs Blas(Custom) Blossom C++11 Chaco DIntegration Dlopen GMP Gmm Kbipack Lapack(Custom) MPI MathEx Mesh Mmg3d NativeFileChooser Netgen ONELAB ONELABMetamodel OptHom Parser Plugins Post Solver TetGen/BR Tetgen1.5 Voro3D
Packaged by      : heltai
Web site         : http://gmsh.info
Mailing list     : gmsh@onelab.info

**Comment by tamiko** [comment]
@luca-heltai Ah, blas, netgen and tetgen support was missing.

**Comment by tamiko** [comment]
@luca-heltai It might be a good idea to eventually improve the `FindGMSH.cmake` script and query for supported modes (in case `DEAL_II_ALLOW_PLATFORMINTROSPECTION` is set to true) and verbosely tell if gmsh misses some crucual configuration.

**Comment by luca-heltai** [comment]
@tamiko, the thing is that I have no idea what modes we need... it is not really well documented what options work with what configuration. We tried a few, and went for using the most robust one. It does not say in the documentation that you need a certain set of features... I'll try to figure out something better for 9.1

**Comment by davydden** [comment]
@tamiko @luca-heltai `create_tria_02` still fails on my tester. `blas` is there in gmsh, but `netgen` and `tetgen` are not, i will try adding them to `gmsh` and see if it fixes the issue. 

Meanwhile, @luca-heltai could you fix this https://github.com/dealii/dealii/issues/5643 ? I don't think `gmsh` dependency is documented, you could also mention what is known to be required for new functionality to work.

**Comment by davydden** [comment]
> Meanwhile, @luca-heltai could you fix this #5643 ?

hm, it looks like it is documented now https://www.dealii.org/developer/index.html maybe just update this text in requirements on gmsh and close the issue?

**Comment by davydden** [comment]
i added a line about Gmsh https://github.com/dealii/dealii/pull/6541 

@luca-heltai 

> Meanwhile, @luca-heltai could you fix this #5643 ?

#5643 is actually about installation instructions, so it's not just what Gmsh needs to be build with. Nevermind my comment.

**Comment by bangerth** [comment]
> @dealii/dealii mhm I am actually tempted to wait a bit for #6537, #6538, #6536, #6535 and add them to the rc4 as well. (And after that bite at everyone who wants to have further PRs that are not critical bugfixes ^^). Opinions?

All four of these are now merged. Only the first changes any code, the rest are probably safe to pull over to the branch, though I don't think the last two really are critical. Either way, let's bring these over and then finally be done! We've all spent enough time on this.

**Comment by tamiko** [comment]
@bangerth My final set of pull-requests is #6549, #6548, #6547. I am happy to tag rc4 immediately after. The testsuite should be fixed now (see updated readme on gmsh requirements).

**Comment by bangerth** [comment]
Tell me what to do with #6548 and we'll do the rc4 and then call it a day.

**Comment by bangerth** [comment]
OK, merged.

Now go with rc4 before anyone changes their mind :-)  (I know that there are still the AD doc patches out there, and also the update for step-18. I also appreciate the work that @jppelteret and @drwells have put into this, but I do think that we need to come to a conclusion at one point and we seem to just be piling up things as we go, without coming to convergence.)

**Comment by davydden** [comment]
@bangerth 

> (I know that there are still the AD doc patches out there, and also the update for step-18. I also appreciate the work that @jppelteret and @drwells have put into this, but I do think that we need to come to a conclusion at one point and we seem to just be piling up things as we go, without coming to convergence.)

i thought the conclusion was to at least add AD documentation https://github.com/dealii/dealii/pull/6542 which @jppelteret promised and delivered. 
Let's not rush out the release just because of 1-2 days difference. 
I am sure if there would be no goal of getting it in 9.0, @jppelteret would invest time to do it right now (as he's busy) and I certainly would not try to help reviewing it at 12 at night.

**Comment by tamiko** [comment]
It would also be worth to consider merging a fix for #6552 (introduced by #6536), or reverting #6536.

**Comment by jppelteret** [comment]
> ...the update for step-18. I also appreciate the work that @jppelteret and @drwells have put into this, but I do think that we need to come to a conclusion...

For #6473 I am going to submit a patch reformatting the table of images today. Our discussion ended with us deciding not to make any code or result modifications.

**Comment by bangerth** [comment]
@davydden and @jppelteret -- fair enough. I do appreciate the work that both of you are putting into this!

**Comment by tamiko** [comment]
@bangerth Let me put it that way - if the three remaining PRs/issues linked above is all that is left, we had a pretty successful release marathon. (With a bit of last-minute fallout, but that's why we have release candidates and bug-fix rounds in the first place.) :-)

**Comment by davydden** [comment]
@dealii/dealii btw, i think we shall have another release task -- update deal.II in package managers (Spack, Candi)

**Comment by bangerth** [comment]
The release task list is in the release-papers git repo -- feel free to add entries there!

**Comment by davydden** [comment]
> The release task list is in the release-papers git repo -- feel free to add entries there!

done. Updated in the original post as well.

**Comment by davydden** [comment]
@tamiko @luca-heltai `create_tria_02.debug` still fails for me even after I added `tetgen` and `netgen` to `gmsh`. @tamiko with what external libraries do you build `gmsh` with?

**Comment by luca-heltai** [comment]
@davydden , are you using the gmsh installation that comes with spack? That's what I used.

**Comment by masterleinad** [comment]
I have
```
$ gmsh -info
Version          : 3.0.6
License          : GNU General Public License
Build OS         : Linux64
Build date       : 20171105
Build host       : gmsh.info
Build options    : 64Bit Ann Bamg Bfgs Blas(Generic) Blossom C++11 Chaco DIntegration Dlopen Fltk Gmm Jpeg(Fltk) Kbipack Lapack(Generic) LinuxJoystick MathEx Med Mesh Metis Mmg3d Mpeg Netgen ONELAB ONELABMetamodel OpenCASCADE OpenGL OptHom PETSc Parser Plugins Png(Fltk) Post SLEPc Solver Taucs TetGen/BR Tetgen1.5 Voro3D Zlib
FLTK version     : 1.3.4
PETSc version    : 3.7.5 (complex arithmtic)
OCC version      : 7.1.0
MED version      : 3.2.0
Packaged by      : gitlab-runner
Web site         : http://gmsh.info
Mailing list     : gmsh@onelab.info
```

**Comment by luca-heltai** [comment]
for me it is:

~~~
691-heltai@turbolo: spack[develop] $ gmsh -info
Version          : 3.0.6
License          : GNU General Public License
Build OS         : MacOSX
Build date       : 20180505
Build host       : turbolo
Build options    : 64Bit Ann Bamg Bfgs Blas(Custom) Blossom C++11 Chaco DIntegration Dlopen GMP Gmm Kbipack Lapack(Custom) MPI MathEx Mesh Mmg3d NativeFileChooser Netgen ONELAB ONELABMetamodel OptHom Parser Plugins Post Solver TetGen/BR Tetgen1.5 Voro3D
Packaged by      : heltai
Web site         : http://gmsh.info
Mailing list     : gmsh@onelab.info
~~~

**Comment by davydden** [comment]
>  are you using the gmsh installation that comes with spack? That's what I used.

yes, including those additions https://github.com/spack/spack/pull/8060  I was failing for me prior to those changes as well...

**Comment by tamiko** [comment]
> @tamiko @luca-heltai `create_tria_02.debug` still fails for me even
> after I added `tetgen` and `netgen` to `gmsh`. @tamiko with what
> external libraries do you build `gmsh` with?

```
Version          : 3.0.6
License          : GNU General Public License
Build OS         : Linux64
Build date       : 20180508
Build host       : kestrel
Build options    : 64Bit Ann Bamg Bfgs Blas Blossom C++11 Cairo Chaco DIntegration Dlopen Fltk GMP Gmm Jpeg Kbipack Lapack LinuxJoystick MathEx Mesh Mmg3d Mpeg NativeFileChooser Netgen ONELAB ONELABMetamodel OpenCASCADE OpenGL OptHom Parser Plugins Png Post Solver TetGen/BR Tetgen1.5 Voro3D Zlib
FLTK version     : 1.3.3
OCC version      : 7.2.0
Packaged by      : portage
Web site         : http://gmsh.info
Mailing list     : gmsh@onelab.info
```

In combination with opencascade-7.2.0

**Comment by davydden** [comment]
this test fails on half of the testers https://cdash.kyomu.43-1.org/testSummary.php?project=1&name=gmsh%2Fcreate_tria_02.debug&date=2018-05-09

**Comment by tamiko** [comment]
@davydden On `tester` with updated gmsh it works fine.

**Comment by davydden** [comment]
> On tester with updated gmsh it works fine.

I also have Blas, netgen, tetgen enabled, but this does not help. Whateve “updated” means, it’s not just that.

I build dealii with the latest OCE, if I recall correctly, but gmsh isn’t build against it

**Comment by tamiko** [comment]
> I build dealii with the latest OCE, if I recall correctly, but gmsh isn’t build against it 

You probably also need gmsh being built against OCE. That's the one
difference I can see right now.

**Comment by tamiko** [comment]
@davydden Any luck to convince gmsh to pass the test? :-)

**Comment by tamiko** [comment]
@masterleinad @davydden @Rombur Give me one more testsuite spin on current `dealii-9.0` head, namely `v9.0.0-rc5-6-ge549f8f0bf`, please :-)

**Comment by bangerth** [comment]
What did you decide about #6542?

**Comment by tamiko** [comment]
@bangerth The consensus so far is to cherry-pick #6542 to the release branch once it is merged - which is about now :-)

**Comment by bangerth** [comment]
OK, we should then be set now!

**Comment by tamiko** [comment]
@luca-heltai @kronbichler @jppelteret Please have a critical look at https://tamiko.kyomu.43-1.org/tmp/doc/

**Comment by jppelteret** [comment]
Thanks @tamiko! I picked up a few small typo's in the documentation I recently wrote. See #6564.

**Comment by davydden** [comment]
@tamiko 

> Any luck to convince gmsh to pass the test? :-)

looks like `gmsh` build with `oce` did the trick, I am rerunning the tester now, give it 4-5 hours to complete (it's a standard oldish desktop PC).

**Comment by bangerth** [comment]
I'd say #6567 falls in the category of "nuisance" and not "critical" at this point in a release cycle.

**Comment by tamiko** [comment]
Regarding our code gallery:
 - Pull request to fix "ceres": https://github.com/dealii/code-gallery/pull/45
 - We get a couple of deprecation warnings: https://github.com/dealii/code-gallery/issues

Apart from this, all examples configure, compile and run fine. (Only checked for functionality, not correctness of the computational result.)

**Comment by luca-heltai** [comment]
@bangerth, le me disagree on #6567. Independently on how we tackle the issue that @code @endcode is used both for inlined code and for example code, I think the indentation issue is worth addressing anyway.

**Comment by bangerth** [comment]
I agree it should be addressed. Just not for the release :-)

**Comment by luca-heltai** [comment]
Why not? If the documentation compiles and aligns correctly with the fix, why not include it?

**Comment by tamiko** [comment]
@bangerth @luca-heltai I am boarding a flight right now. Personally, I am inclined to merge #6567 in order to have better readable documentation. I will board a flight now and am back this afternoon (eastern time).

**Comment by luca-heltai** [comment]
Let's compromise: lets merge *without* specifying a different entry in the stylesheet. The difference style can by postponed to 9.1. But at least indentation looks ok in all tutorials.

**Comment by luca-heltai** [comment]
I see your point. I'm generating the documentation with the updated version, and with the old version. I'll post here the diff between the html directories.

**Comment by tamiko** [comment]
@kronbichler This [failing test](https://cdash.kyomu.43-1.org/testDetails.php?test=10270945&build=1676) on node27 looks odd.

**Comment by luca-heltai** [comment]
@bangerth, 

> It has been wrong for several years apparently, without any of us or any of our users ever noticing. 

Once you know that there is a stain on the wall, your eyes always fall there, don't they? That's what's happening to me now...

**Comment by kronbichler** [comment]
> This failing test on node27 looks odd.

I've seen this failure also sporadically on other machines as well. I've looked into it a bit and the problem is a race condition. While this sounds scary, I have verified that it cannot happen in the standard case of discontinuous basis functions and face integrals, which is the case the extension was written for. However, something goes wrong in the connectivity computation for the unusual case that we have:
- integrals on interior faces
- continuous finite element bases
- threading with a medium (but not too large) computational domain, i.e., a relatively high number of threads as compared to elements.

I would be leaned to simply ignore the issue for the 9.0 release (i.e., simply reduce the test to what we know works and maybe add a text of warning in the documentation of the MatrixFree class) and create a proper fix for 9.1. It does not make much sense mathematically to compute face integrals on a continuous basis because they ought to be zero. Overall, the TBB threading support for DG face integrals is not comprehensive yet anyway, so I'd not risk any code changes at this point.

**Comment by bangerth** [comment]
On 05/11/2018 05:56 AM, Luca Heltai wrote:
> Once you know that there is a stain on the wall, your eyes always fall there, 
> don't they?

They have a habit of doing that, don't they...

**Comment by kronbichler** [comment]
Just to be clear, I would not push on adding #6577 to the release branch if it incurs any delay. It merely helps in having a clean and stable test suite. I decided last weekend - when I first saw this test to fail sporadically - to not rush for a last minute change, so I'm certainly not keen on a last second change at this point.

**Comment by drwells** [comment]
I am in favor of including #6577 (or writing a separate PR that moves it, for now, into the failing test directory) so that we can get clean test suite runs.

@tamiko Will there be another release candidate soon?

**Comment by tamiko** [comment]
@dealii/dealii Let's do two last cherry picks. @drwells No more release candidate - I will generate the documentation and verify that Luca's change didn't break anything and rerun one configuration to verify that Martins' change works and after that tag a release.

**Comment by tamiko** [comment]
@kronbichler Thanks a lot for the detailed analysis! :-)

**Comment by tamiko** [comment]
@luca-heltai Release step 17a/ instructs me to ask you for Mac packages ;-)

**Comment by tamiko** [comment]
@davydden @drwells @tjhei *ping* we need virtualbox images, spack and aur updates :-)

**Comment by drwells** [comment]
I can handle the AUR update. I will have it done in the next few days.

**Comment by tamiko** [comment]
@masterleinad I wrote an e-mail regarding the online documentation for the web server.

For the time being a preview can be found [here](https://tamiko.kyomu.43-1.org/tmp/doc/).

**Comment by tamiko** [comment]
@bangerth @tjhei May I ask you to do the announcements on aspect-devel@, na digest, cig-all@, num.info@ and siam-cse@?

**Comment by luca-heltai** [comment]
@tamiko , on it right now. I have a slow connection, so the package will take some time to upload...

**Comment by drwells** [comment]
I have a revised preprint that gets rid of the marginalia and some of the lines that extend into the margins, but I do not have website access. I attached it here.

[paper.pdf](https://github.com/dealii/dealii/files/1997081/paper.pdf)

I would be quite appreciative if someone could put it up, but Matthias has more important things to do now :)

**Comment by davydden** [comment]
> @davydden @drwells @tjhei ping we need virtualbox images, spack and aur updates

I marked Spack as done some 30 min after you tagged 9.0 ;-) (PR is there)

**Comment by tamiko** [comment]
> I have a revised preprint that gets rid of the marginalia and some of
> the lines that extend into the margins, but I do not have website
> access. I attached it here.
>
> [paper.pdf](https://github.com/dealii/dealii/files/1997081/paper.pdf)
>
> I would be quite appreciative if someone could put it up, but Matthias
> has more important things to do now :)

Done.

**Comment by tamiko** [comment]
Also docker images: https://hub.docker.com/r/dealii/dealii/

**Comment by luca-heltai** [comment]
Doing docker images now.

**Comment by drwells** [comment]
I have updated the AUR instructions. I also updated the wikipedia page.

**Comment by tamiko** [comment]
@dealii/candi-devs *ping* candi update :-)

**Comment by tamiko** [comment]
@dealii/dealii All, thanks a lot for two crazy productive weeks! I will close this issue now (with 4 minor points open that will be resolved in the near future).

**Comment by davydden** [comment]
> @dealii/candi-devs ping candi update :-)

Judging from commit history in candi fork, we need to ping @tjhei for that.

**Comment by bangerth** [comment]
@tjhei is in Nowhere, Sweden and may or may not have internet connectivity. If nobody else can do this in his stead, it may simply have to wait. (It seems like everyone took the end of the semester as an occasion to get going somewhere. @tamiko is/was in Philadelphia IIRC, and I'm in and out a lot of different places in Beijing and the rest of the country for nearly 4 weeks...)

**Comment by koecher** [comment]
I’ll do this as soon as possible. But I’m also abroad on a conference in the US. I’ll. do it from next week on when I’m back in my office in Hamburg. Thanks for the remainder.

**Comment by drwells** [comment]
The AUR package has been updated and (aside from trilinos, gmsh, and adolc, which I do not have installed at the moment) I can verify that all quick tests pass and we can build with GCC 8.1 without warnings.

**Comment by bangerth** [comment]
On 05/12/2018 07:02 AM, Matthias Maier wrote:
> May I ask you to do the announcements on aspect-devel@, na digest, cig-all@, 
> num.info@ and siam-cse@?

I've sent things to aspect-devel and cig-all. I've removed num.info from the 
tasks list -- this was the list used by the Rannacher group, but he's retired 
and none of us are immediate members of that research group any more. I 
haven't been on SIAM-CSE in a long while and don't know whether it's still 
active. But I signed up again and will check whether it's worth sending to.

That leaves NA Digest. This is a high-visibility mailing list a lot of people 
get. I think you (@tamiko) should post the announcement there to get the 
visibility you deserve. It's also going to be good to get your name out in the 
community. If we all take turns sending announcements there, then that's good 
for everyone.

**Comment by tamiko** [comment]
> But I signed up again and will check whether it's worth sending to.

I have already done SIAM-CSE. It's a moderated list with reasonably high
volume. Will do the other one tomorrow. :-)

**Comment by bangerth** [comment]
> I have already done SIAM-CSE.

OK, great. I signed up but did not hear back from them yet.
